### DISCUSSION TITLE
Help Flashing code

### DISCUSSION THREAD

**Discussion by Rage65** [SammyIAm/Moppy2#208] [Q&A] [answered]
i downloaded the arduino code is main.cpp the code I want to flash on to the arduino? I appriciate any help!

**Comment by samphonic** [comment]
If you're using the Arduino IDE, you should open *only* the `.ino` file (I think anyway, it's been a long time since I used it).

If you're using a different IDE (or in any case really) the file that gets flashed is a compiled binary containing all the code (not just main.cpp), but how to generate that will vary (but it's not unique or special to Moppy, so the documentation or Google search for your IDE should help out)

**Reply by Rage65** [reply]
What about compiling the gui for controlling the moppy

**Reply by samphonic** [reply, marked as answer]
Are you trying to *compile* it, or do you just want to launch it and use it?  If the latter, just [follow the instructions on the wiki](https://github.com/Sammy1Am/Moppy2/wiki/Getting-Started#controller-setup).

If the former: It's been a *long* time since I worked on the code, so I don't remember everything.  Once again, the [instructions on the wiki](https://github.com/Sammy1Am/Moppy2/wiki/Getting-Started-with-Moppy-Development#control-software-the-pc-part) are a great place to start.  Might be a little out of date, but in short, something Java with Gradle should do it.

**Reply by Rage65** [reply]
Thx I guess I did not see that part or did not remember it
